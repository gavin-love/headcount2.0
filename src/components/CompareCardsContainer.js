import React from "react";
import Card from "./Card";
import PropTypes from 'prop-types';
import '../styles/CompareCardContainer.css'

const CompareCardsContainer = ({ selectedDistricts }) => {
  if (selectedDistricts) {
    const allSelectedDistricts = selectedDistricts.map(
      (selectedDistrict, index) => {
        if (selectedDistrict.locationOne) {
          return (
            <Card
              locationOne={selectedDistrict.locationOne}
              locationOneAverage={selectedDistrict.locationOneAverage}
              comparedAverage={selectedDistrict.comparedAverage}
              locationTwo={selectedDistrict.locationTwo}
              locationTwoAverage={selectedDistrict.locationTwoAverage}
              key={index}
              findAverages={selectedDistrict.findAverages}
            />
          );
        }

        return (
          <Card
            findAverages={selectedDistrict.findAverages}
            location={selectedDistrict.location}
            stats={selectedDistrict.stats}
            key={index}
          />
        );
      }
    );

    return <div className="compare-cards-container">{allSelectedDistricts}</div>;
  } else {
    return <div />;
  }
};

export default CompareCardsContainer;
