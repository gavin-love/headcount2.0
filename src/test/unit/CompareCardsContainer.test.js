import React from "react";
import ReactDOM from "react-dom";
import CompareCardsContainer from "../../components/CompareCardsContainer.js";
import { shallow } from "enzyme";

describe("CompareCardContainer", () => {
  let wrapper;
  let mockData;
  let mockFunction;

  beforeEach(() => {
    mockFunction = jest.fn();
    mockData = [{average: 34, findAverages: {mockFunction}, location: "colorado", stats: { 2004: 0.24,                  2005: 0.63, 2006: 0.11 }},
                {average: 12, findAverages: {mockFunction}, location: "academy 20", stats: { 2012: 0.24, 2013: 0.63, 2014: 0.11 }},
                {comparedAverage: 0.46, locationOne: "COLORADO", locationOneAverage: 0.53, locationTwo: "ACADEMY 20", locationTwoAverage: 0.40 }
              ];
  });

  it("matches the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should", () => {
    wrapper = shallow(
      <CompareCardsContainer selectedDistricts={mockData} />
    );

    expect(wrapper).toMatchSnapshot();
  });

});