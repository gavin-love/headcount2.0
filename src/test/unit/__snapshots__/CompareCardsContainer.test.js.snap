// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompareCardContainer matches the snapshot 1`] = `undefined`;

exports[`CompareCardContainer should 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CompareCardsContainer
    selectedDistricts={
        Array [
            Object {
              "average": 34,
              "findAverages": Object {
                "mockFunction": [Function],
              },
              "location": "colorado",
              "stats": Object {
                "2004": 0.24,
                "2005": 0.63,
                "2006": 0.11,
              },
            },
            Object {
              "average": 12,
              "findAverages": Object {
                "mockFunction": [Function],
              },
              "location": "academy 20",
              "stats": Object {
                "2012": 0.24,
                "2013": 0.63,
                "2014": 0.11,
              },
            },
            Object {
              "comparedAverage": 0.46,
              "locationOne": "COLORADO",
              "locationOneAverage": 0.53,
              "locationTwo": "ACADEMY 20",
              "locationTwoAverage": 0.4,
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          findAverages={
                    Object {
                              "mockFunction": [Function],
                            }
          }
          location="colorado"
          selected={undefined}
          stats={
                    Object {
                              "2004": 0.24,
                              "2005": 0.63,
                              "2006": 0.11,
                            }
          }
/>,
        <Card
          findAverages={
                    Object {
                              "mockFunction": [Function],
                            }
          }
          location="academy 20"
          selected={undefined}
          stats={
                    Object {
                              "2012": 0.24,
                              "2013": 0.63,
                              "2014": 0.11,
                            }
          }
/>,
        <Card
          comparedAverage={0.46}
          findAverages={undefined}
          locationOne="COLORADO"
          locationOneAverage={0.53}
          locationTwo="ACADEMY 20"
          locationTwoAverage={0.4}
/>,
      ],
      "className": "compare-cards-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "findAverages": Object {
            "mockFunction": [Function],
          },
          "location": "colorado",
          "selected": undefined,
          "stats": Object {
            "2004": 0.24,
            "2005": 0.63,
            "2006": 0.11,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "findAverages": Object {
            "mockFunction": [Function],
          },
          "location": "academy 20",
          "selected": undefined,
          "stats": Object {
            "2012": 0.24,
            "2013": 0.63,
            "2014": 0.11,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "function",
        "props": Object {
          "comparedAverage": 0.46,
          "findAverages": undefined,
          "locationOne": "COLORADO",
          "locationOneAverage": 0.53,
          "locationTwo": "ACADEMY 20",
          "locationTwoAverage": 0.4,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            findAverages={
                        Object {
                                    "mockFunction": [Function],
                                  }
            }
            location="colorado"
            selected={undefined}
            stats={
                        Object {
                                    "2004": 0.24,
                                    "2005": 0.63,
                                    "2006": 0.11,
                                  }
            }
/>,
          <Card
            findAverages={
                        Object {
                                    "mockFunction": [Function],
                                  }
            }
            location="academy 20"
            selected={undefined}
            stats={
                        Object {
                                    "2012": 0.24,
                                    "2013": 0.63,
                                    "2014": 0.11,
                                  }
            }
/>,
          <Card
            comparedAverage={0.46}
            findAverages={undefined}
            locationOne="COLORADO"
            locationOneAverage={0.53}
            locationTwo="ACADEMY 20"
            locationTwoAverage={0.4}
/>,
        ],
        "className": "compare-cards-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "findAverages": Object {
              "mockFunction": [Function],
            },
            "location": "colorado",
            "selected": undefined,
            "stats": Object {
              "2004": 0.24,
              "2005": 0.63,
              "2006": 0.11,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "findAverages": Object {
              "mockFunction": [Function],
            },
            "location": "academy 20",
            "selected": undefined,
            "stats": Object {
              "2012": 0.24,
              "2013": 0.63,
              "2014": 0.11,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "function",
          "props": Object {
            "comparedAverage": 0.46,
            "findAverages": undefined,
            "locationOne": "COLORADO",
            "locationOneAverage": 0.53,
            "locationTwo": "ACADEMY 20",
            "locationTwoAverage": 0.4,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
